eox_core/api/v1/tests/integration/test_views.py

```diff
@@ -370,6 +431,595 @@ def test_update_user_in_tenant_user_not_found(self, param: str, value: str) -> N
+        self.create_enrollment(self.tenant_x, enrollment_data)
+        enrollment_data["is_active"] = True
+        enrollment_data["mode"] = "honor"
+
+        response = self.update_enrollment(self.tenant_x, data=enrollment_data)
+
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response_data["user"], user_data["username"])
+        self.assertEqual(response_data["course_id"], enrollment_data["course_id"])
+        self.assertEqual(response_data["mode"], enrollment_data["mode"])
+        self.assertTrue(response_data["is_active"])
+
+    @ddt.data("email", "username")
+    def test_update_valid_enrollment_update_invalid_mode(self, param: str) -> None:
+        """
+        Test updating an existing enrollment. Update to invalid mode.
+
+        Open edX definitions tested:
+        - `get_edxapp_user`
+        - `update_enrollment`
+
+        Expected result:
+        - The status code is 400.
+        - The response contains an error message about the mode not found.
+        """
+        user_data = next(FAKE_USER_DATA)
+        enrollment_data = {
+            param: user_data[param],
+            "course_id": self.course_id,
+            "is_active": True,
+            "mode": self.mode,
+        }
+        self.create_user(self.tenant_x, user_data)
+        self.create_enrollment(self.tenant_x, enrollment_data)
+        enrollment_data["mode"] = "masters"
+
+        response = self.update_enrollment(self.tenant_x, data=enrollment_data)
+
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertIn("non_field_errors", response_data)
+        self.assertEqual(response_data["non_field_errors"], ["Mode not found"])
+
+    @ddt.data("email", "username")
+    def test_update_enrollment_does_not_exist(self, param: str) -> None:
+        """
+        Test updating an enrollment that does not exist in a tenant.
+
+        Open edX definitions tested:
+        - `get_edxapp_user`
+        - `update_enrollment`
+
+        Expected result:
+        - The status code is 202.
+        - The response contains an error message about the enrollment not found.
+        """
+        user_data = next(FAKE_USER_DATA)
+        self.create_user(self.tenant_x, user_data)
+        enrollment_data = {
+            param: user_data[param],
+            "course_id": self.course_id,
+            "is_active": False,
+            "mode": "honor",
+        }
+
+        response = self.update_enrollment(self.tenant_x, data=enrollment_data)
+
+        response_data = response.json()
+        self.assertEqual(response.status_code, status.HTTP_202_ACCEPTED)
+        self.assertEqual(response_data["error"]["detail"], f"No enrollment found for {user_data['username']}")
+
+    @ddt.data("email", "username")
+    def test_update_valid_enrollment_force_post(self, param: str) -> None:
```

Comment:
Same question here about what's the difference between updating and force updating.

Reply:
The difference is explained here. To update the initial data we must use `force = True`
https://github.com/eduNEXT/eox-core/blob/2708af49afb32ce19bbcd02a17c1d87e24846af8/eox_core/api/v1/views.py#L599-L602

Reply:
Done: https://github.com/eduNEXT/eox-core/commit/490a19eee770c832882cb730a705843911370df9